### PR TITLE
add recipe for infra-agent on amazon linux 2022

### PR DESCRIPTION
**Pull request by brushknight**
- add AL2022 recipe (Amazon Linux 2022)
- merge Al2 and AL2022 into the same recipe
- rename old amazonlinux.yml -> amazonlinux_1.yml
- rename old amazonlinux2.yml -> amazonlinux_2.yml

### CODE REVIEW

File: recipes/newrelic/infrastructure/amazonlinux.yml [hunk 1]
```diff
@@ -107,6 +115,17 @@ install:
             echo "touch is required to run the newrelic install. Please install touch and re-run the installation." >&2
             exit 15
           fi
+        - |
+          IS_INFRA_AVAILABLE=$(curl -Is {{.NEW_RELIC_DOWNLOAD_URL}}infrastructure_agent/linux/yum/amazonlinux/{{.AMAZON_LINUX_VERSION}}/{{.ARCH}}/newrelic-infra.repo | grep " 2[0-9][0-9] " | wc -l)
+          if [ $IS_INFRA_AVAILABLE -eq 0 ] ; then
```
Comment (rubenruizdegauna) [inline recipes/newrelic/infrastructure/amazonlinux.yml:120]: What does this grep `2[0-9][0-9]` do? ☝️

**Comment by Julien4218** [comment]
@brushknight would you have more details around this change? Any JIRA ticket perhaps?
What is AL2022?
I'll pass this on to the team to review. Side note, do you know that renaming the file to `_old.yml` will still keep this file as a valid recipe and be used for installation? If you intend this recipe file to be used, could I suggest to find a better name to indicate that this is indeed a valid recipe and what it targets to use, versus the other amazon linux file. Perhaps even better, do we really need 2 amazon files, or could we combine them in 1?

File: recipes/newrelic/infrastructure/amazonlinux.yml (hunk 1 above)
Comment (brushknight) [reply recipes/newrelic/infrastructure/amazonlinux.yml:120]: We curl just headers and look for the 2xx response code.

Comment (rubenruizdegauna) [reply recipes/newrelic/infrastructure/amazonlinux.yml:120]: Oh, I didn't get the headers detail 🤦

**Comment by brushknight** [comment]
We are adding support for Amazon Linux 2022. It is not possible to merge into the same repo as for Amazon Linux "1" we use el6 repo and for Amazon Linux 2/2022 we have amazonlinux repos.

**Comment by rubenruizdegauna** [comment]
Hi @amolero-nr 👋  Requested changes where applied. Will you be able to take a look 🙏 ? Thanks!!

File: recipes/newrelic/infrastructure/amazonlinux.yml [hunk 2]
```diff
@@ -11,11 +11,19 @@ installTargets:
...
+    platformFamily: rhel
+    platformVersion: "2022"
+
 
 keywords:
   - Infrastructure
   - Agent
-  - Amazon Linux
+  - Amazon Linux 2
+  - Linux2
```
[the remark this answers is not in the document]
Comment (rubenruizdegauna) [reply recipes/newrelic/infrastructure/amazonlinux.yml]: Hi @amolero-nr I'll change that one, sorry I missed it

[the remark this answers is not in the document]
Comment (rubenruizdegauna) [reply recipes/newrelic/infrastructure/amazonlinux.yml]: done :)

[the remark this answers is not in the document]
Comment (amolero-nr) [reply recipes/newrelic/infrastructure/amazonlinux.yml]: No worries, thanks @rubenruizdegauna!

**Comment by amolero-nr** [comment]
I double-checked with my team earlier today and the direction agreed was to wait for the code freeze to be lifted next Wednesday. I will merge/deploy this PR then...

**Comment by brushknight** [comment]
@amolero-nr I had to push force as there were conflicts and it was easier just to redo commits

**Review by amolero-nr — DISMISSED** [review]
Approving for now, will also explore options to consolidate both Amazon recipes into one.

**Pull request merged**